sdk/python/kfp/_client.py

```diff
@@ -113,6 +113,11 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client
 
   def _load_config(self, host, client_id, namespace, other_client_id, other_client_secret):
     config = kfp_server_api.configuration.Configuration()
+
+    host = host or ''
+    # Preprocess the host endpoint to prevent some common user mistakes.
+    host = host.lstrip(r'(https|http)://').rstrip('/')
```

Comment:
I vaguely remember there were scenarios where this had to be an actual URI with non-empty path.
Something like `http://1.2.3.4/something/#pipeline`

Reply:
Do you mean we have to reserve the prefix `http://`?